src/drive/targets/public/index.jsx

```diff
@@ -127,58 +127,55 @@ const init = async () => {
-                        />
-                      )}
-                    />
-                    {models.file.shouldBeOpenedByOnlyOffice(data) && (
-                      <Redirect from="/" to={`onlyoffice/${data.id}`} />
-                    )}
-                  </>
-                )}
-                <Route
-                  path="/"
-                  component={() => <LightFileViewer files={[data]} />}
-                />
-              </Route>
-            </Router>
-          ) : (
-            <Router history={hashHistory}>
-              <Route component={PublicLayout}>
-                <Redirect from="/files/:folderId" to="/folder/:folderId" />
-                <Route path="folder(/:folderId)" component={PublicFolderView}>
-                  <Route path="file/:fileId/revision" component={FileHistory} />
-                </Route>
-                {isOnlyOfficeEnabled() && (
-                  // Used to open an only office file inside a shared folder
+          <Router history={hashHistory}>
+            <Route component={PublicLayout}>
+              {isOnlyOfficeEnabled() && (
+                <>
                   <Route
                     path="onlyoffice/:fileId"
                     component={props => (
                       <OnlyOfficeView
                         {...props}
                         isPublic={true}
-                        isInSharedFolder={true}
+                        username={username}
+                        isFromSharing={isOnlyOfficeDocShared}
+                        isInSharedFolder={!isFile}
                       />
                     )}
                   />
-                )}
-              </Route>
-              <Route path="external/:fileId" component={ExternalRedirect} />
-              <Redirect from="/*" to={`folder/${sharedDocumentId}`} />
-            </Router>
-          )}
+                  {models.file.shouldBeOpenedByOnlyOffice(data) && (
+                    <Redirect from="/" to={`onlyoffice/${data.id}`} />
+                  )}
+                </>
+              )}
+
+              {isFile && (
+                <Route
+                  path="/"
+                  component={() => <LightFileViewer files={[data]} />}
+                />
+              )}
+
+              {!isFile && (
+                <>
+                  <Redirect from="/files/:folderId" to="/folder/:folderId" />
+                  <Route path="folder(/:folderId)" component={PublicFolderView}>
+                    <Route
+                      path="file/:fileId/revision"
+                      component={FileHistory}
+                    />
+                  </Route>
+                </>
+              )}
+            </Route>
+
+            {!isFile && (
+              <>
+                <Route path="external/:fileId" component={ExternalRedirect} />
+                <Redirect from="/*" to={`folder/${sharedDocumentId}`} />
+              </>
+            )}
+          </Router>
         </App>,
```

Comment:
Is it really necessary? Is it clearer that way? When I see "+59 −47" for a "mutualisation" I'm having trouble. 

There is a lot of test on `isFile` and I find it hard to understand :/. I prefer the read of the previous version.

Reply:
Je trouvais ça plus simple dans le sens où les oublis/régressions étaient dues au fait que j'ai oublié le cas folder. Et l'autre chose c'est qu'il y a doublon entre le folder et le file. Perso je trouve ça plus lisible comme ça, d'avoir une seule lecture et de me dire que pendant la lecture, il y a des cas particuliers, plutôt que d'avoir 2 niveaux de lecture.

L'idée était de se rapprocher de l'approute. D'ailleurs on n'est pas obligé de conditionner la plupart des routes finalement (la seule importante étant celle qui récupère le `/`), le seul risque c'est d'avoir une erreur en tapant une url à la main non ?

On peut très bien revenir en arrière ceci dit :)